docs/src/HandlingFiles.md

```diff
@@ -0,0 +1,136 @@
+
+For the common task of reading files sequentially (from the beginning) at a particular path, FS2 provides two convenience methods, depending on your need for asynchronous operations.
+
+## Synchronous file reading
+
+```tut:book
+io.file.readAll[Task](Paths.get("build.sbt"), 1024)
+```
+
+## Asynchronous file reading
+
+```tut:book
+io.file.readAllAsync[Task](Paths.get("build.sbt"), 1024)
+```
+
+These close the file after it is no longer needed.
+
+# Writing files sequentially
+
+Similarly, two methods are provided which offer an easy way to write to
+files sequentially from the beginning.
+
+## Synchronous file writing
+
+```tut:book
+io.file.writeAll[Task](Files.createTempFile("xxx", "yyy"))
+```
+
+## Asynchronous file writing
+
+```tut:book
+io.file.writeAllAsync[Task](Files.createTempFile("xxx", "yyy"))
+```
+
+Again, these close the file when it is no longer needed.
+
+# Example 1: Reading a UTF-8 file into a stream of `String`s
+
+```tut:book
+val source = io.file.readAll[Task](Paths.get("build.sbt"), 1024)
+
+source.through(text.utf8Decode)
+```
+
+# Example 2: Copying a file
+
+```tut:book
+val destination = io.file.writeAll[Task](Paths.get("build.sbt.backup"))
+
+source.to(destination).run.run
+```
+
+# Example 3: Copying a file, converting to lower-case
+
+```tut:book
+source.
+  through(text.utf8Decode).
+  map(_.toLowerCase).
+  through(text.utf8Encode).
+  to(destination).
+  run.run
+```
+
+# The `FileHandle` trait
+
+The `FileHandle` trait provides access to read or modify an "open" file.
+As it is based on the features from Java NIO, it allows random access
+to read and/or write, and also allows locking of all, or parts of, the file.
+It can also be used to force synchronization of a file to disk, and request
+information on the file's size.
+
+## Obtaining a FileHandle
+
+A `FileHandle` can be obtained using a `java.nio.file.Path`, a `java.nio.channel.FileChannel`,
+or a `java.nio.channel.AsynchronousFileChannel` object. This `FileHandle` is wrapped in a `Pull`,
+and the appropriate resources will be closed when the `Pull` is complete.
+
+### Code examples
+
+```tut:book
+io.file.pulls.fromPath[Task](Paths.get("build.sbt"), List(StandardOpenOption.READ))
+io.file.pulls.fromPathAsync[Task](Paths.get("build.sbt"), List(StandardOpenOption.WRITE))
+```
+
+We can create a channel from a `FileInputStream` or `FileOutputStream`:
+
+```tut:book
+io.file.pulls.fromFileChannel[Task](
+  Task.delay(
+    new java.io.FileInputStream(new java.io.File("build.sbt")).getChannel
+  )
+)
+```
+We can also open files asynchronously with the `io.file.pulls.fromAsynchronousFileChannel` call.
+
+## The `Pull` monad and `FileHandle` operations
+
+Giving the ability to manipulate files within the `Pull` monad provides
+a lot of power. It is instructive to look at one of the simpler implementations, [pulls.readAllFromFileHandle](https://github.com/functional-streams-for-scala/fs2/tree/topic/redesign/io/src/main/scala/fs2/io/file/pulls.scala).
+
```

Comment:
Use relative links like is done in guide.md please.
On Tue, May 17, 2016 at 5:58 AM Gary Coady <notifications@github.com> wrote:

> In docs/src/HandlingFiles.md
> <https://github.com/functional-streams-for-scala/fs2/pull/618#discussion_r63495236>
> :
>
> > +We can create a channel from a `FileInputStream` or `FileOutputStream`:
> > +
> > +```tut:book
> > +io.file.pulls.fromFileChannel[Task](
> > +  Task.delay(
> > +    new java.io.FileInputStream(new java.io.File("build.sbt")).getChannel
> > +  )
> > +)
> > +```
> > +We can also open files asynchronously with the `io.file.pulls.fromAsynchronousFileChannel` call.
> > +
> > +## The `Pull` monad and `FileHandle` operations
> > +
> > +Giving the ability to manipulate files within the `Pull` monad provides
> > +a lot of power. It is instructive to look at one of the simpler implementations, [pulls.readAllFromFileHandle](https://github.com/functional-streams-for-scala/fs2/tree/topic/redesign/io/src/main/scala/fs2/io/file/pulls.scala).
> > +
>
> The file should exist after this commit.
>
> Yes, an example would be good, I'll add something.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/functional-streams-for-scala/fs2/pull/618/files/5b9c206e3034859902ac612d2eeab90893297895#r63495236>
>